lib/services/database/comment/comment_repository_service.dart

```diff
@@ -0,0 +1,115 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/comment/comment_data.dart";
+import "package:proxima/models/database/comment/comment_firestore.dart";
+import "package:proxima/models/database/comment/comment_id_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/models/database/user_comment/user_comment_data.dart";
+import "package:proxima/models/database/user_comment/user_comment_firestore.dart";
+import "package:proxima/services/database/comment/post_comment_repository_service.dart";
+import "package:proxima/services/database/comment/user_comment_repository_service.dart";
+import "package:proxima/services/database/firestore_service.dart";
+
+/// This service allows to interact with the comments of the application.
+/// It will allow to add, delete and retrieve comments from the firestore database.
+/// And it will handle the references between the comments and the users.
+class CommentRepositoryService {
```

Comment:
That's a good point indeed.
The two futures from `addComment` cannot be awaited in parallel because we need the result of `PostCommentRepository.addComment` to create the `UserCommentFirestore` to add afterward.

I now await the two futures in parallel in `deleteComment`, done in 5a70a13004e1964be15b78fb120a86772bd84382.

I believe the futures in `deleteAllComments` are already awaited in parallel but I have noticed that the code could be made more conscise. I do that in cef755a7eedb407045eb4de879b25d34c55d9567.

Reply:
Good job :)